docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,264 @@
+# Ingesting and Utilizing RAG Artifacts Design Proposal
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+## 1. Introduction
+Purpose of this document is to propose `ilab` CLI changes to extend the InstructLab flow to generate and utilize RAG artifacts
+in the customer workflows. The proposal includes new commands to ingest a local RAG from pre-processed customer documentation, that we assume to be generated from docling output, and updates to the existing commands to utilize these RAG artifacts.
+
+## 2. Use Case
+* Accelerate POCs of fine-tuning LLMs + RAG.
+* Generate artifacts that data scientists can use to build custom solutions in their development environment:
+  * Jupyter Notebooks & Python Scripts
+  * Podman Desktop
+  * Red Hat OpenShift AI
+  * Internal Red Hat CI systems for products or services (e.g., Lightspeed products)
+
+## 3. Proposed Commands
+**Note**: In the context of version 1.4, currently under development, no changes to the command-line interface should be
+allowed. Therefore, we also propose alternative approaches to run the same RAG pipelines using existing `ilab` commands or
+other provided tools.
+
+### 3.1 RAG Ingestion Pipeline Command
```

Comment:
How will this be handled for fine tuning? 

I would expect that it ends up that many of the components end up being re-invented here.

Reply:
Are you asking about fine-tuning of the embedding model or the response-generation model or both?

Reply:
Both.